resources/deployments/mongo/device_deployments.go

```diff
@@ -433,7 +433,7 @@ func (d *DeviceDeploymentsStorage) DecommissionDeviceDeployments(deviceId string
 	_, err := session.DB(DatabaseName).C(CollectionDevices).UpdateAll(selector, update)
```

Comment:
my understanding of this comment:
https://github.com/go-mgo/mgo/blob/9a2573d4ae52a2bf9f5b7900a50e2f8bcceeb774/session.go#L2488

is that no, it's not possible. especially:
` It is not an error for the update to not be applied on any documents because the selector doesn't match.` 

this would be the only reason to return mgo.ErrNotFound.